syntax = "proto3";

import "types.proto";

import "google/protobuf/empty.proto";
import "google/protobuf/wrappers.proto";
import "google/protobuf/timestamp.proto";
import "google/protobuf/duration.proto";

message BasicInfo{
    Address Addr = 1;
    google.protobuf.Timestamp RegTime = 2;
    string ServerSign = 3;
}

message CloudCheckResult{
    string ServerSign = 1;
    string DataHash = 2;
    bool CheckResult = 3;
}

message NodeCheckRecord{
    bool IsUploaded = 1;
    string DataHash = 2;
    map<string, int32> ReciteList = 3;
    int32 Result = 4;
}

message CarPosCheckRecord{
    string DataHash = 1;
    map<string, int32> CheckResult = 3;
}

message CloudCheckRecord{
    string ServerSign = 1;
    string DataHash = 2;
    bool Result = 3;
}

message InitializeInput{
    BasicInfo Info = 1;
    NodeList AdjInfo = 2;
}

message UpdateAdjInput{
    NodeList AdjInfo = 1;
}

message ReciteInput{
    Address To = 1;
    bool Result = 2;
    int64 Round = 3;
}

message PosCheckResult{
    map<string, bool> Results = 1;
    string DataHash = 2;
    int64 Round = 3;
    Address To = 4;
}

message RoundResult{
    map<string, NodeCheckRecord> NodeCheckList = 2;
    map<string, CloudCheckRecord> CloudCheckList = 3;
    map<string, CarPosCheckRecord> CarPosCheckList = 4;
}

message DataHashSet{
    repeated string HashSet = 1;
}

message CloudCheckInput{
    Address To = 1;
    string ServerSign = 2;
    string DataHash = 3;
    bool Result = 4;
    int64 Round = 5;
}

message RoundInfoInput{
    map<string, int32> NodeResult = 1;
    NodeList NodeList = 2;
    NodeList CloudList = 3;
    NodeList PositiveList = 4;
}

message LongTermCacheInput{
    string DataHash = 1;
}

message GetDataHashInput{
    google.protobuf.Timestamp StartTime = 1;
    google.protobuf.Timestamp EndTime = 2;
}

message StatusResult{
    int64 Round = 1;
    repeated Address Nodes = 2;
}

message NodeList{
    repeated Address Nodes = 1;
}